src/irmin-pack/unix/checks.ml

```diff
@@ -50,6 +50,17 @@ let path =
 
 let deprecated_info = (Cmdliner.Term.info [@alert "-deprecated"])
 
+let ppf_or_null ppf =
+  let null =
+    match Sys.os_type with
+    | "Unix" | "Cygwin" -> "/dev/null"
+    | "Win32" -> "NUL"
+    | _ -> invalid_arg "invalid os type"
```

Comment:
Perhaps we can use https://ocaml.org/p/ocaml-variants/4.12.0%2Bdomains/doc/Stdlib/Format/index.html#val-make_formatter to define a generic "null" formatter that doesn't depend on the OS?

Reply:
Hmm, yeah I can look into it. This code is just copy-pasta of what already existed. :sweat_smile:

Reply:
haha yeah I saw that but it still looks icky :P

Reply:
Since this isn't the goal of this current PR, I opened an issue for future enhancement. https://github.com/mirage/irmin/issues/2231